next/src/modules/adminPage/components/SubTable.tsx

```diff
@@ -0,0 +1,304 @@
+    }
+
+    const { selectedOrganizations: organizationIds } = params as {
+      selectedOrganizations: string[];
+    };
+
+    onAddData(mainId, organizationIds);
+  };
+
+  const dataSource = isMembershipType
+    ? data.map((item) => {
+        if ('user' in item) {
+          const { user } = item as OrganizationMembership;
+          return {
+            key: user.id,
+            firstName: user.firstName,
+            lastName: user.lastName,
+            email: user.email,
+          };
+        }
+        return {};
+      })
+    : isDatasetAccessType
+    ? data.map((item) => {
+        if ('dataset' in item) {
+          const { dataset } = item as DatasetAccess;
+          return {
+            key: dataset.id,
+            datasetName: dataset.name,
+          };
+        }
+        return {};
+      })
+    : data.map((item) => {
+        if ('organization' in item) {
+          const { organization } = item;
+          return {
+            key: organization.id,
+            organizationName: organization.name,
+            firstDatasetName: organization.datasetAccess[0]?.dataset.name,
+            totalDatasetLeft:
+              organization.datasetAccess.length &&
+              organization.datasetAccess.length - 1,
+          };
+        }
+        return {};
+      });
+
+  const existedIds = isMembershipType
+    ? data.map((membership: OrganizationMembership) => membership.user.id)
+    : isDatasetAccessType
+    ? data.map((datasetAccess: DatasetAccess) => datasetAccess.dataset.id)
+    : data.map((organizationMemberships: OrganizationMembership) => {
+        return organizationMemberships.organization.id;
+      });
+
+  // The filteredAllData array is created by filtering the allData array to exclude already existing data entries.
+  const filteredAllData = isMembershipType
+    ? allData.filter((user: User) => !existedIds.includes(user.id))
+    : isDatasetAccessType
+    ? allData.filter(
+        (dataset: Dataset) =>
+          // And only dataset isActive
+          !existedIds.includes(dataset.id) && dataset.isActive,
+      )
+    : allData.filter(
+        (organization: Organization) => !existedIds.includes(organization.id),
+      );
+
+  return (
+    <>
+      <AddingForm
+        type={type}
+        isOpen={addView}
+        allData={filteredAllData}
+        onClose={handleCloseCreateView}
+        onAdd={handleAdd}
+      />
+
+      <Collapse>
+        <Collapse.Panel
+          header={
+            isMembershipType
+              ? 'Organization Memberships'
+              : isDatasetAccessType
+              ? 'Dataset Access'
+              : 'Organizations'
+          }
+          key="1"
+        >
+          <Button
+            type="primary"
+            onClick={handleOpenCreateView}
+            style={{ marginBottom: 16 }}
+          >
+            {isMembershipType
+              ? 'Add New Membership'
+              : isDatasetAccessType
+              ? 'Add New Dataset'
```

Comment:
```suggestion
              ? 'Add Dataset Access'
```

I think this is more accurate